### PR TITLE
Fixes to the CSM logs

### PR DESCRIPTION
**Pull request by mew2057**
Fixed JSON generation for transaction log:
Issue #498 
Issue #502

Fixed transaction log rotation and filebeats implementation:
Issue #507

### CODE REVIEW

**Comment by NickyDaB** [comment]
What was the original problem? How do the changes here approach to fix this problem?

File: csmconf/csm_master.cfg [hunk 1]
```diff
@@ -12,7 +12,7 @@
             "consoleLog"                        :   false,
             "sysLog"                            :   true,
             "fileLog"                           :   "/var/log/ibm/csm/csm_master.log",
-            "#rotationSize_comment_1"           :   "Maximum size (in bytes) of the log file, 1000000000 bytes is ~1GB",
+            #rotationSize_comment_1"           :   "Maximum size (in bytes) of the log file, 1000000000 bytes is ~1GB",
```
Comment (besawn) [inline csmconf/csm_master.cfg]: I am 90% sure that this change breaks the configuration file and prevents the daemon from even starting...

Comment (mew2057) [reply csmconf/csm_master.cfg]: I thought this change was removed, fixing the file.

**Comment by mew2057** [comment]
> What was the original problem? How do the changes here approach to fix this problem?
@NickyDaB 

There were two discrete problems:

1. JSON generation was failing for compute nodes where the number was a multiple of 256. This was an issue with the pointer de reference  which was effectively performing a `modulo 256` on the number of nodes. (#502, #498)

2. The transactional (and metrics) log was growing and not rotating. This was resulting in a rather large file on customer systems (#507). This basically indicates that the rollover that had been implemented was non functional.

@besawn I tested these to see if I could get the existing schema to work, but to no good results. I think filebeats may interfere with the clean operation of the boost rollover. This my primary hypothesis for what happened on the customer site, as I couldn't get the traditional rollover to  function.

I adopted the N rollover for two reasons:
1. Preservation of old transactional data in the event of a BDS or filebeat outage. This will allow the transactional tables to be rebuilt wholesale and have no data loss.
2. Circumventing the rollover issue found in [filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/how-filebeat-works.html):


> There is a limitation to Filebeat’s at-least-once delivery guarantee involving log rotation and the deletion of old files. 
> If log files are written to disk and rotated faster than they can be processed by Filebeat, or if files are deleted while the output is unavailable, data might be lost. 
> On Linux, it’s also possible for Filebeat to skip lines as the result of inode reuse. See Frequently asked questions for more details about the inode reuse issue.

**Pull request merged**